flink-ml-lib/src/main/java/org/apache/flink/ml/feature/minmaxscaler/MinMaxScaler.java

```diff
@@ -0,0 +1,165 @@
+public class MinMaxScaler
+        implements Estimator<MinMaxScaler, MinMaxScalerModel>, MinMaxScalerParams<MinMaxScaler> {
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    public MinMaxScaler() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public MinMaxScalerModel fit(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        DataStream<Tuple2<DenseVector, DenseVector>> minMaxVectors =
+                computeMinMaxVectors(tEnv.toDataStream(inputs[0]), getFeaturesCol());
+        DataStream<MinMaxScalerModelData> modelData = genModelData(minMaxVectors);
+        MinMaxScalerModel model =
+                new MinMaxScalerModel().setModelData(tEnv.fromDataStream(modelData));
+        ReadWriteUtils.updateExistingParams(model, getParamMap());
+        return model;
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.saveMetadata(this, path);
+    }
+
+    public static MinMaxScaler load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        return ReadWriteUtils.loadStageParam(path);
+    }
+
+    /**
+     * Generates minMax scaler model data.
+     *
+     * @param minMaxVectors Input distributed minMaxVectors.
+     * @return MinMax scaler model data.
+     */
+    private static DataStream<MinMaxScalerModelData> genModelData(
+            DataStream<Tuple2<DenseVector, DenseVector>> minMaxVectors) {
+        DataStream<MinMaxScalerModelData> modelData =
+                DataStreamUtils.mapPartition(
+                        minMaxVectors,
+                        new RichMapPartitionFunction<
+                                Tuple2<DenseVector, DenseVector>, MinMaxScalerModelData>() {
+                            @Override
+                            public void mapPartition(
+                                    Iterable<Tuple2<DenseVector, DenseVector>> dataPoints,
+                                    Collector<MinMaxScalerModelData> out) {
+                                DenseVector minVector = null;
+                                DenseVector maxVector = null;
+                                int vecSize = 0;
+                                for (Tuple2<DenseVector, DenseVector> dataPoint : dataPoints) {
+                                    if (maxVector == null) {
+                                        vecSize = dataPoint.f0.size();
+                                        maxVector = dataPoint.f1;
+                                        minVector = dataPoint.f0;
+                                    }
+                                    for (int i = 0; i < vecSize; ++i) {
```

Comment:
nits: vecSize could be replaced with `maxVector.size()`

Reply:
ok